File: tp_architecture_logicielle_avancee/python/cours.py
class Cours:
    def __init__(self, nom, professeur):
        self.nom = nom
        self.professeur = professeur

    def afficher_details(self):
        print(f"Cours : {self.nom}, Professeur : {self.professeur}")


class LegacyCoursSystem:
    def get_cours_string(self):
        return "Mathématiques;Dupont"


class CoursAdapter:
    def __init__(self, legacy_system):
        self._legacy = legacy_system

    def get_cours(self):
        data = self._legacy.get_cours_string()
        nom, professeur = data.split(";")
        return Cours(nom, professeur)

File: tp_architecture_logicielle_avancee/java/main/Cours.java
public class Cours {
    private String nom;
    private String professeur;

    public Cours(String nom, String professeur) {
        this.nom = nom;
        this.professeur = professeur;
    }

    public String getNom() {
        return nom;
    }

    public String getProfesseur() {
        return professeur;
    }

    public void afficherDetails() {
        System.out.println("  Cours : " + nom + ", Professeur : " + professeur);
    }
}
